### DISCUSSION TITLE
Typst in text citation

### DISCUSSION THREAD

**Discussion by alxsimon** [quarto-dev/quarto-cli#8981] [Feature Requests]
### Description

From what I can tell, the handling of the different citation styles has not been made in the Typst template yet?
I'd be happy to try at least implementing the simplest case of in text citation:
`@ref` should give a `#cite(<ref>, form: "prose")` in Typst, in contrast with the normal `[@ref]`.
Any pointer to how this should be implemented is welcome, or an explanation on how that works in Latex so that it can be ported in Typst?
Thanks

**Comment by mcanouil** [comment]
This is done by Pandoc (citeproc) not Quarto.

**Comment by cderv** [comment]
Pandoc 3.1.12 is the version that supports this `form:` attributes. 

See release note at https://pandoc.org/releases.html#pandoc-3.1.12-2024-02-14

> Improve citation support ([#9452](https://github.com/jgm/pandoc/issues/9452)). Emit form: "prose" or form: "year" qualifiers if the citation is author-in-text or suppress-author. Strip initial comma from suffix, since typst will add an extra one.

Done in this commit : https://github.com/jgm/pandoc/commit/aa00714ba476c8558eb897003fe69009a02b0171 

When we will update to 3.1.12, this will be possible in Quarto too.

**Reply by alxsimon** [reply]
Great! Thanks for the details.
